usr.sbin/adduser/adduser.sh

```diff
@@ -613,6 +665,47 @@ get_password() {
 	fi
 }
 
+# get_zfs_support
+#	Determine if ZFS is supported on this system.
+#
+get_zfs_support()
+{
+	_zfs_list=`${ZFSCMD} list 2>&1`
+	if [ "$?" -ne 0 ]; then
+		zsupported=
+	elif [ "$_zfs_list" = "no datasets available" ]; then
+		zsupported=
+	else
+		zsupported="yes"
+	fi
+}
+
+# create_zfs_dataset
+#   Create ZFS dataset owned by the user that was just added.
+#
+create_zfs_dataset() {
+	${ZFSCMD} create -p -o "mountpoint=${uhome}" ${zfsopt} "${zhome}"
```

Comment:
DO NOT set mountpoint, unless it's absolutely necessary.  This would complicate replication to a different location (because ZFS would copy mountpoint to a foreign ZFS dataset).